canonical-web-and-design/maas-ui
# refactor(ui): replace useHistory with useNavigate

## Done

- Replace useHistory with useNavigate as a step towards React Router v6.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that the different list searches update the url (e.g. machine, device, lxd, subnets etc.)
- Check that submitting forms or pressing cancel returns to the previous page.
- Create a new user and login and check that you get redirected to the intro.

## Fixes

Fixes: #3960.
